### PR TITLE
fix kubectl swagger cache bug

### PR DESCRIPTION
**Pull request by adohe-zz**
Please refer #21353 for more detail.
@nikhiljindal ptal

### CODE REVIEW

**Comment by AdoHe** [comment]
@k8s-bot test this #22088

**Comment by nikhiljindal** [comment]
@k8s-bot test this github issue: https://github.com/kubernetes/kubernetes/issues/22088

**Comment by AdoHe** [comment]
@lavalamp kindly ping.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

File: pkg/api/errors/errors.go [hunk 1]
```diff
@@ -92,6 +92,14 @@ func FromObject(obj runtime.Object) error {
 	return &UnexpectedObjectError{obj}
 }
 
+// TypeNotFoundError is returned when specified type
+// can not found in schema
+type TypeNotFoundError string
```
Comment (nikhiljindal) [inline pkg/api/errors/errors.go]: This file contains api errors generated by the server.
I dont think that this is the right place to define this error (which is used only by kubectl).

I think it is better to move it to pkg/api/validation/schema.go

**Comment by nikhiljindal** [comment]
Thanks! Looks great, just one comment to move the error definition.

**Comment by AdoHe** [comment]
Just update this.

**Comment by AdoHe** [comment]
@nikhiljindal kindly ping.

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Pull request merged**